clang/lib/DPCT/MigrateCmakeScript.cpp

```diff
@@ -604,3 +566,93 @@ std::string convertCmakeCommandsToLower(const std::string &InputString) {
 
   return OutputStream.str();
 }
+
+static void doCmakeScriptAnalysis() {
+
+  for (auto &Entry : CmakeScriptFileBufferMap) {
+    auto &Buffer = Entry.second;
+
+    // Collect varible name and its value
+    parseVariable(Buffer);
+  }
+}
+
+static void unifyInputFileFormat() {
+  for (auto &Entry : CmakeScriptFileBufferMap) {
+    auto &Buffer = Entry.second;
+
+    // Convert input file to be LF
+    bool IsCRLF = fixLineEndings(Buffer, Buffer);
+    ScriptFileCRLFMap[Entry.first] = IsCRLF;
+
+    // Convert cmake command to lower case in cmake script files
+    Buffer = convertCmakeCommandsToLower(Buffer);
+  }
+}
+
+static void applyCmakeMigrationRules() {
+  for (auto &Entry : CmakeScriptFileBufferMap) {
+    auto &Buffer = Entry.second;
+
+    // Apply implicit migration rules
+    applyImplicitMigrationRules(Buffer);
+
+    // Apply user define migration rules
```

Comment:
please comment on:  when yaml based rule is used,  or used for what kind cases.

Reply:
ImplicitMigrationRules is used when the migration logic is difficult to be described with yaml based rule syntax. 
Currently, only migration of cmake_minimum_required() is applied with ImplicitMigrationRule. Once ImplicitMigrationRule for some cmake syntax is implemented, the yaml based rules for the same cmake syntax will be skipped and a msg is emitted to notify the user.